x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/NodeDeprecationChecks.java

```diff
@@ -542,4 +550,24 @@ static DeprecationIssue checkMonitoringExporterPassword(
         final String url = "https://www.elastic.co/guide/en/elasticsearch/reference/7.7/monitoring-settings.html#http-exporter-settings";
         return new DeprecationIssue(DeprecationIssue.Level.CRITICAL, message, url, details, null);
     }
+
+    static DeprecationIssue checkClusterRoutingAllocationIncludeRelocationsSetting(final Settings settings,
+                                                                                   final PluginsAndModules pluginsAndModules,
+                                                                                   final ClusterState clusterState) {
+        DeprecationIssue nodeDeprecationIssue = getClusterRoutingAllocationIncludeRelocationsSettingDeprecationIssue(settings);
+        if (nodeDeprecationIssue != null) {
+            return nodeDeprecationIssue;
+        }
+
+        // The setting is dynamic so it can be defined stored in the ClusterState settings
+        return getClusterRoutingAllocationIncludeRelocationsSettingDeprecationIssue(clusterState.metadata().settings());
+    }
+
+    private static DeprecationIssue getClusterRoutingAllocationIncludeRelocationsSettingDeprecationIssue(Settings settings) {
+        return checkRemovedSetting(settings,
+            CLUSTER_ROUTING_ALLOCATION_INCLUDE_RELOCATIONS_SETTING,
+            "https://www.elastic.co/guide/en/elasticsearch/reference/master/migrating-8.0.html#breaking_80_allocation_changes",
+            DeprecationIssue.Level.WARNING
```

Comment:
I think it's `CRITICAL` at least in the case where it's in the node settings. If it's set dynamically then that's technically ok to proceed, we'll just archive it, but that's also a case we can fix at runtime so maybe not so bad to call it `CRITICAL` too? Will the upgrade assistant automatically help users remove this setting if set dynamically or do we need to take action on that front too?

Reply:
I've changed it to `CRITICAL` when it's in the node settings. I think we could tackle the case when the setting is set dynamically with the upgrade assistant, we just need to open a ticket in Kibana describing how the assistant can mitigate the problem. I'll open the ticket.